hircluster.h

```diff
@@ -136,7 +137,7 @@ typedef struct redisClusterContext {
     sslInitFn *ssl_init_fn; /* Func ptr for SSL context initiation */
 
     void (*on_connect)(const struct redisContext *c, int status);
-    void (*event_callback)(const struct redisClusterContext *cc, int event);
+    void *event_callback;
```

Comment:
We can use an opaque struct here, i.e. we put only a typedef in .h and define the struct in .c.

```c
// in hircluster.h
typedef struct cluster_event_callback redisClusterEventCallback;

// in hircluster.c
struct cluster_event_callback { ... };
```

In this way, we can avoid casting, without exposing the internals.

```suggestion
    redisClusterEventCallback *event_callback;
```